tasks/shared_libraries/database/shared_db.py

```diff
@@ -182,7 +182,7 @@ def execute_connection_with_error_handling(engine, sql: str, parameters: Dict):
                 "Failed to execute the query due to {err}. Retrying {retry} time(s)"
             )
             logger.error(message, err=err, retry=retry + 1)
-            INITIAL_BACKOFF_IN_SECONDS = exponential_delay(
```

Comment:
Not sure if this is correct yet logic wise, I would expect you would want to initialize backoff_in_seconds before the loop and then use it as an input to the function to get the new time value.

Reply:
changed to

```
    delay = INITIAL_BACKOFF_IN_SECONDS
    for retry in range(RETRIES + 1):
        try:
            with engine.begin() as connection:
                connection.execute(sql, parameters)
            break
        except exc.OperationalError as err:
            message = (
                "Failed to execute the query due to {err}. Retrying {retry} time(s)"
            )
            logger.error(message, err=err, retry=retry + 1)
            delay = exponential_delay(
                delay, BACKOFF_FACTOR
            )
            continue
    else:
        message = f"Failed to execute the query after {RETRIES} retries."
        logger.error(message)
        raise Exception(message)
```